Detectors/TPC/qc/include/TPCQC/PID.h

```diff
@@ -62,18 +65,11 @@ class PID
   /// Dump results to a file
   void dumpToFile(std::string filename);
 
-  /// get 1D histograms
-  std::vector<TH1F>& getHistograms1D() { return mHist1D; }
-  const std::vector<TH1F>& getHistograms1D() const { return mHist1D; }
-
-  /// get 2D histograms
-  std::vector<TH2F>& getHistograms2D() { return mHist2D; }
-  const std::vector<TH2F>& getHistograms2D() const { return mHist2D; }
-
+  /// return a map of vectors of histograms
+  std::map<std::string_view, std::vector<TH1*>> getMapOfHisto() { return mapHist2D; }
+  const std::map<std::string_view, std::vector<TH1*>> getMapOfHisto() const { return mapHist2D; }
  private:
-  std::vector<TH1F> mHist1D{};
-  std::vector<TH2F> mHist2D{};
-
+  std::map<std::string_view, std::vector<TH1*>> mapHist2D;
```

Comment:
I guess the class owns the histograms. The destructor should delete them.
```suggestion
  std::map<std::string_view, std::vector<std::unique_ptr<TH1>>> mapHist2D;
```
Requires `#include <memory>`